### PR TITLE
Fix `torch.dsplit` docs dim specification

### LINKED ISSUE #70445: Error in `torch.dsplit` documentation
### 📚 The doc issue

In the documentation of `torch.dsplit`, the sentence "calling torch.tensor_split(input, indices_or_sections, dim=2) (the split dimension is 1)" is error. Because the split dimension of `dsplit` is 2 instead of 1.

By contrast, in the documentation of `torch.hsplit`, it says "calling torch.tensor_split(input, indices_or_sections, dim=0) (the split dimension is 0)", which is correct.

### Suggest a potential alternative/fix

calling torch.tensor_split(input, indices_or_sections, dim=2) (the split dimension is 2)

cc @brianjo @mruberry @albanD @soulitzer

### PR DESCRIPTION
**Pull request by jaketae**
Fixes https://github.com/pytorch/pytorch/issues/70445.

### CODE REVIEW

**Review by mruberry — APPROVED** [review]
Thanks @jaketae!
